harvard-lts/mirador-pdiiif-plugin
# Update tooling and publication workflow

**JIRA Ticket**: [LTSVIEWER-228](https://jira.huit.harvard.edu/browse/LTSVIEWER-228)

- [PR for mps-viewer](https://github.com/harvard-lts/mps-viewer/pull/58), which contains a version bump and documentation updates for plugin development

# What does this Pull Request do?

This replaces Parcel with Rollup for providing the ESM file to use when the package is published. It also brings the ability to preview and work on the plugin locally using Webpack and Webpack dev server to bundle a stock Mirador viewer, the plugin and Mirador initialisation code. A Github action workflow is included, which will allow the plugin to be published to NPM whenever a new release is created in Github. This must be configured first by following instructions in the linked PR.  

# How should this be tested?

- Checking out this branch and following the instructions in the README, should result in you being able to run `npm run serve` to view the plugin in a local version of Mirador.

**The Github workflow cannot be tested until this is merged as it will need to run against main AND be configured with the appropriate secrets**

- Once merged and configured, the action can be run by following the instructions in the linked [mps-viewer RP](https://github.com/harvard-lts/mps-viewer/pull/58)

# Test coverage

Yes/No: Are changes in this pull-request covered by:

- unit tests? No
- integration tests? No

# Interested parties

@enriquediaz
